### PR TITLE
Remove empty staging directories after insert into hive table

### PR DESCRIPTION
**Pull request by haozhun**
Supersedes #3851 because I had to make some additional changes to tests.

Thank you @kokosing

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java [hunk 1]
```diff
@@ -802,6 +802,36 @@ public void finishInsert(ConnectorSession session, ConnectorInsertTableHandle in
...
+                FileSystem fileSystem;
+                try {
+                    fileSystem = hdfsEnvironment.getFileSystem(session.getUser(), writePathRoot.get());
+                }
+                catch (IOException e) {
+                    throw new PrestoException(HIVE_FILESYSTEM_ERROR, e);
+                }
+                for (PartitionUpdate partitionUpdate : partitionUpdates) {
+                    verify(!partitionUpdate.getTargetPath().equals(partitionUpdate.getWritePath()));
+                    Path writePath = new Path(partitionUpdate.getWritePath());
```
Comment (kokosing) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: What about this commit: https://github.com/prestodb/presto/pull/3851/commits/79d2bafb6d1f8792ca5cb47c88509dcda75d9dfb

Havining this, you woundn't need to wrap wrap `writePath` as `Path` each time. IMO  this commit makes `PartitionUpdate` be coherent with `LocatioService` as in that case both would return  `Path` for `wrtiePath` and `targetPath`. Moreover  it is more intuitive to return `Path` for paths.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java [hunk 2]
```diff
@@ -812,6 +842,19 @@ public void finishInsert(ConnectorSession session, ConnectorInsertTableHandle in
         clearRollback();
     }
 
+    private boolean isSameOrParent(Path parent, Path child)
```
Comment (kokosing) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: make it static

**Comment by kokosing** [comment]
@haozhun Thank you for taking care of this.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java (hunk 2 above)
Comment (haozhun) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: Good catch

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java (hunk 1 above)
Comment (haozhun) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: Good catch. Updated.

**Comment by kokosing** [comment]
Superseded by #5352

**Pull request closed without merging**